src/main/java/com/aws/greengrass/deployment/templating/TransformerWrapper.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.aws.greengrass.deployment.templating;
+
+import com.amazon.aws.iot.greengrass.component.common.ComponentRecipe;
+import com.aws.greengrass.util.Pair;
+import com.fasterxml.jackson.databind.JsonNode;
+
+import java.io.IOException;
+import java.lang.reflect.InvocationTargetException;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.net.URLClassLoader;
+import java.nio.file.Path;
+import java.util.List;
+
+public class TransformerWrapper {
+
+    RecipeTransformer transformer;
+
+    /**
+     * Function to execute expansion.
+     * @param pathToExecutable  the jar to run.
+     * @param className         the name of the parser class.
+     * @param template          the template recipe file.
+     * @throws ClassNotFoundException       if the jar does not contain the parser class.
+     * @throws IllegalTransformerException  if the class does not implement the RecipeTransformer interface.
+     * @throws InstantiationException       if constructor is not public.
+     * @throws IllegalAccessException       idk...
+     * @throws RecipeTransformerException   for everything else.
+     */
+    public TransformerWrapper(Path pathToExecutable, String className, ComponentRecipe template,
+                              JsonNode templateConfig)
+            throws RecipeTransformerException, ClassNotFoundException, IllegalTransformerException,
+            NoSuchMethodException, InvocationTargetException, InstantiationException, IllegalAccessException {
+        if (!pathToExecutable.toFile().exists()) {
+            throw new RecipeTransformerException("Could not find template parsing jar to execute");
+        }
+        System.out.println(pathToExecutable);
+        URLClassLoader loader;
+        try {
+            loader = new URLClassLoader(new URL[] {pathToExecutable.toFile().toURI().toURL()},
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code contains a resource that might not be closed properly. This can cause a resource leak that slows down or crashes your system.

**Fix**
Consider closing the following resource: *loader*. The resource is referenced at line 50. The resource is closed at line 61. There are other execution paths that do not contain closure statements, for example, when Class.forName throws an exception. Ensure *loader* is closed in a try-finally block or declared in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).